### PR TITLE
Added more templates to benchmarking

### PR DESCRIPTION
**Pull request by LiamRiddell**

### CODE REVIEW

**Comment by SebastianStehle** [comment]
Can we unify the testrunner and the benchmark? I use the testrunner for dottrace and so on. Perhaps just an argument to run them without benchmark.net?

File: Mjml.Net.Benchmarking/TemplateBenchmarks.cs [hunk 1]
```diff
@@ -4,25 +4,56 @@ namespace Mjml.Net.Benchmarking
 {
     public class TemplateBenchmarks
     {
-        private readonly MjmlRenderer MjmlRenderer;
-        public string TemplateAmario { get; set; }
+        private readonly MjmlRenderer MjmlRenderer;
```
Comment (SebastianStehle) [inline Mjml.Net.Benchmarking/TemplateBenchmarks.cs:7]: I would write a params source here:

https://benchmarkdotnet.org/articles/features/parameterization.html#sample-introparamssource

**Review by SebastianStehle — CHANGES_REQUESTED** [review]

File: Mjml.Net.Benchmarking/TemplateBenchmarks.cs (hunk 1 above)
Comment (LiamRiddell) [reply Mjml.Net.Benchmarking/TemplateBenchmarks.cs:7]: I'll update this.

Comment (LiamRiddell) [reply Mjml.Net.Benchmarking/TemplateBenchmarks.cs:7]: We'll lose the naming from the output of the benchmark. Is that what we want?

Comment (SebastianStehle) [reply Mjml.Net.Benchmarking/TemplateBenchmarks.cs:7]: The name from the output? You can loop over the fields and provide the file name as param and then load the actual content in the setup.

File: Mjml.Net.Benchmarking/TemplateBenchmarks.cs [hunk 2]
```diff
@@ -5,84 +5,60 @@ namespace Mjml.Net.Benchmarking
...
-        public string TemplateSphero { get; set; }
-        public string TemplateManyHeroes { get; set; }
+     
+        [ParamsSource(nameof(MjmlTemplates))]
+        public string MjmlTemplateFileName { get; set; }
 
-        public TemplateBenchmarks()
-        {
-            MjmlRenderer = new();
+        public IEnumerable<string> MjmlTemplates => new[] {
```
Comment (SebastianStehle) [inline Mjml.Net.Benchmarking/TemplateBenchmarks.cs:12]: I would just loop over the directory => Directory.GetFiles("*.mjml").Select(x => x.Name);  or so

Comment (SebastianStehle) [reply Mjml.Net.Benchmarking/TemplateBenchmarks.cs:12]: The point is that we can easily add more templates and do not have to change the code for that.

**Comment by LiamRiddell** [comment]
> Can we unify the testrunner and the benchmark? I use the testrunner for dottrace and so on. Perhaps just an argument to run them without benchmark.net?

I'll do this now.

**Comment by LiamRiddell** [comment]
@SebastianStehle, I renamed the `Mjml.Net.Benchmarking` to `Mjml.Net.Benchmark`. I think this has affected the NuGet deployment. We should probably not publish `Mjml.Net.Benchmarking` to NuGet anyway?

**Comment by SebastianStehle** [comment]
In the action I only pack Mjml.Net, so it should not be a problem. But it also tries to publish for pull requests. I have tried to fix that now. Feel free to many any changes to the build.

File: Mjml.Net.Benchmark/Program.cs [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+        }
+
+        public static void Main(string[] args)
+        {
+            Parser.Default.ParseArguments<Options>(args)
+                     .WithParsed(o =>
+                     {
+                         if (o.TestRunner)
+                         {
+                             TestRunner.Run();
```
Comment (SebastianStehle) [inline Mjml.Net.Benchmark/Program.cs:21]: Where is the TestRunner coming from?

Comment (SebastianStehle) [reply Mjml.Net.Benchmark/Program.cs:21]: Btw: Can we also use the dynamic approach in the TestRunner?

Comment (LiamRiddell) [reply Mjml.Net.Benchmark/Program.cs:21]: It's static class `TestRunner` which I copied the code from TestRunner project. https://github.com/SebastianStehle/mjml-net/blob/Additional-Benchmark-Templates/Mjml.Net.Benchmark/TestRunner.cs

Comment (LiamRiddell) [reply Mjml.Net.Benchmark/Program.cs:21]: > Btw: Can we also use the dynamic approach in the TestRunner?

I'll do that now :)

**Comment by LiamRiddell** [comment]
@SebastianStehle, take a look now. I've added an new command line options so you can specify the number of iterations. 

```
CMD > Mjml.Net.Benchmark.exe -p -i 100
```

It will iterate each template 100 times compiling.

**Comment by SebastianStehle** [comment]
Is this done?

**Comment by LiamRiddell** [comment]
> Is this done?

I'll fix the conflicts then we can merge :)

**Pull request merged**